src/Sylius/Component/Core/Formatter/StringInflector.php

```diff
@@ -29,7 +30,7 @@ public static function nameToCode(string $value): string
 
     public static function nameToSlug(string $value): string
     {
-        return str_replace(['_'], '-', self::nameToLowercaseCode($value));
+        return str_replace(['_'], '-', self::nameToLowercaseCode(Transliterator::transliterate($value)));
```

Comment:
a change like this does not be a BC break?

Reply:
@GSadee @lchrusciel would you have a look at it? I did it because we are not really resolving slug sometimes. In the case I have found, I had special characters in my string ( '!' in my PR ) which was not erased from it as it should be. Returned value should be from "string!_string!" to "string-string" but it returns "string!-string!". When data provider is looking for it, it is not available to find it with this slug.